### PR TITLE
V22F-453 update for integration tests

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Added extra check to ensure worker thread is running

### CODE REVIEW

File: IPCLib/Socket.cpp [hunk 1]
```diff
@@ -23,6 +23,11 @@ bool ReceivingThread::HasReceivedMessage() const
     return m_received;
 }
 
+bool ReceivingThread::StartedReceiving() const
```
Comment (shoekzema) [inline IPCLib/Socket.cpp:26]: documentation

File: IPCLib/Socket.cpp [hunk 2]
```diff
@@ -85,11 +91,16 @@ void Socket::ReceiveDataAsync()
...
     m_externalReceive = true;
+    while (!m_receivingThread->StartedReceiving() && !m_receivingThread->HasReceivedMessage())
+    {
+        std::this_thread::yield();
+    }
 }
 
 /// @brief Receive data by waiting until data has been written on the socket
-void Socket::ReceiveData()
+void Socket::ReceiveData(bool* p_started)
```
Comment (shoekzema) [inline IPCLib/Socket.cpp:101]: update documentation

File: IPCTests/SocketTest.cpp [hunk 3]
```diff
@@ -373,4 +377,40 @@ TEST(SocketTests, ReceivingThreadThrow)
     std::this_thread::sleep_for(std::chrono::seconds(1));
     server.~ServerSocket();
     ASSERT_EQ(client.AwaitData(buffer, 20), IPCLIB_RECEIVE_ERROR);
+}
+
+#define AMOUNT_OF_TESTS 50000
```
Comment (shoekzema) [inline IPCTests/SocketTest.cpp:382]: to the top

File: IPCTests/SocketTest.cpp [hunk 4]
```diff
@@ -373,4 +377,40 @@ TEST(SocketTests, ReceivingThreadThrow)
...
+void ExhaustionClientThreadSide()
```
Comment (shoekzema) [inline IPCTests/SocketTest.cpp:383]: documentation

File: IPCTests/SocketTest.cpp [hunk 5]
```diff
@@ -373,4 +377,40 @@ TEST(SocketTests, ReceivingThreadThrow)
...
+    char buffer[20];
+    for (int i = 0; i < AMOUNT_OF_TESTS; i++)
+    {
+        ASSERT_DURATION_LE(2, client.AwaitData(buffer, 20));
+        ASSERT_DURATION_LE(2, client.ReceiveDataAsync());
+        ASSERT_EQ(client.SendData("Hi", 3), IPCLIB_SUCCEED);
+    }
+}
+
+TEST(SocketTests, ExhaustionTest)
```
Comment (shoekzema) [inline IPCTests/SocketTest.cpp:397]: documentation

**Review by shoekzema — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]

**Review by shoekzema — APPROVED** [review]

**Pull request merged**